godot does not compile on macOS 10.15 with XCode 11.0

**Godot version:**
bdbf2e818..197b65f32  master


**OS/device including version:**
macOS Catalina 10.15 Dev Prev - MacBook Pro 15'' mid 2017

**Issue description:**
Just out of curiosity, I ran the scons build process on Catalina and encounter the following issue.
On Mojave everything built fine.

Maybe this helps to get godot ready for the next macOS release?!

```
scons: Reading SConscript files ...
Configured with: --prefix=/Applications/Xcode-beta.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
YASM is necessary for WebM SIMD optimizations.
WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!
Checking for C header file mntent.h... no
scons: done reading SConscript files.
scons: Building targets ...
[  1%] Compiling ==> platform/osx/crash_handler_osx.mm
[  1%] build_gles2_headers(["drivers/gles2/shaders/canvas.glsl.gen.h"], ["drivers/gles2/shaders/canvas.glsl"])
[  2%] build_gles3_headers(["drivers/gles3/shaders/canvas.glsl.gen.h"], ["drivers/gles3/shaders/canvas.glsl"])
[  2%] Compiling ==> platform/osx/os_osx.mm
platform/osx/os_osx.mm:1396:2: warning: multiple methods named 'center' found
      [-Wobjc-multiple-method-names]
        [window_object center];
        ^~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSCollectionViewCompositionalLayout.h:597:19: note: 
      using
@property NSPoint center;
                  ^~~~~~
/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:312:1: note: 
      also found
- (void)center;
^~~~~~~~~~~~~~~
1 warning generated.
[  2%] Compiling ==> platform/osx/godot_main_osx.mm
[  2%] Compiling ==> platform/osx/semaphore_osx.cpp
[  2%] Compiling ==> platform/osx/dir_access_osx.mm
[  2%] Compiling ==> platform/osx/joypad_osx.cpp
[  3%] Compiling ==> platform/osx/power_osx.cpp
[  3%] Compiling ==> main/default_controller_mappings.gen.cpp
[  3%] Compiling ==> main/input_default.cpp
[  5%] Compiling ==> main/main.cpp
[  5%] Compiling ==> main/main_timer_sync.cpp
[  5%] Compiling ==> main/performance.cpp
[  5%] Linking Static Library ==> main/libmain.osx.tools.64.a
Ranlib Library         ==> main/libmain.osx.tools.64.a
[  5%] Compiling ==> main/tests/test_astar.cpp
[  5%] Compiling ==> main/tests/test_gdscript.cpp
[  5%] Compiling ==> main/tests/test_gui.cpp
[  5%] Compiling ==> main/tests/test_main.cpp
[  5%] Compiling ==> main/tests/test_math.cpp
[  5%] Compiling ==> main/tests/test_oa_hash_map.cpp
[  6%] Compiling ==> main/tests/test_ordered_hash_map.cpp
[  6%] Compiling ==> main/tests/test_physics.cpp
[  6%] Compiling ==> main/tests/test_physics_2d.cpp
[  6%] Compiling ==> main/tests/test_render.cpp
[  6%] Compiling ==> main/tests/test_shader_lang.cpp
[  6%] Compiling ==> main/tests/test_string.cpp
[  6%] Linking Static Library ==> main/tests/libtests.osx.tools.64.a
Ranlib Library         ==> main/tests/libtests.osx.tools.64.a
[  6%] Compiling ==> modules/register_module_types.gen.cpp
[  7%] Compiling ==> thirdparty/assimp/code/BaseImporter.cpp
[  7%] Compiling ==> thirdparty/assimp/code/BaseProcess.cpp
[  7%] Compiling ==> thirdparty/assimp/code/Bitmap.cpp
[  7%] Compiling ==> thirdparty/assimp/code/CInterfaceIOWrapper.cpp
[  7%] Compiling ==> thirdparty/assimp/code/CalcTangentsProcess.cpp
[  7%] Compiling ==> thirdparty/assimp/code/ComputeUVMappingProcess.cpp
[  7%] Compiling ==> thirdparty/assimp/code/ConvertToLHProcess.cpp
[  7%] Compiling ==> thirdparty/assimp/code/CreateAnimMesh.cpp
[  7%] Compiling ==> thirdparty/assimp/code/DeboneProcess.cpp
[  7%] Compiling ==> thirdparty/assimp/code/DefaultIOStream.cpp
[  7%] Compiling ==> thirdparty/assimp/code/DefaultIOSystem.cpp
[  8%] Compiling ==> thirdparty/assimp/code/DefaultLogger.cpp
[  8%] Compiling ==> thirdparty/assimp/code/DropFaceNormalsProcess.cpp
[  8%] Compiling ==> thirdparty/assimp/code/EmbedTexturesProcess.cpp
[  8%] Compiling ==> thirdparty/assimp/code/Exporter.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXAnimation.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXBinaryTokenizer.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXConverter.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXDeformer.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXDocument.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXDocumentUtil.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXExportNode.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXExportProperty.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXExporter.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXImporter.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXMaterial.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXMeshGeometry.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXModel.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXNodeAttribute.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXParser.cpp
[  8%] Compiling ==> thirdparty/assimp/code/FBXProperties.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FBXTokenizer.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FBXUtil.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FIReader.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FindDegenerates.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FindInstancesProcess.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FindInvalidDataProcess.cpp
[  9%] Compiling ==> thirdparty/assimp/code/FixNormalsStep.cpp
[  9%] Compiling ==> thirdparty/assimp/code/GenFaceNormalsProcess.cpp
[  9%] Compiling ==> thirdparty/assimp/code/GenVertexNormalsProcess.cpp
[  9%] Compiling ==> thirdparty/assimp/code/Importer.cpp
[  9%] Compiling ==> thirdparty/assimp/code/ImporterRegistry.cpp
[  9%] Compiling ==> thirdparty/assimp/code/ImproveCacheLocality.cpp
[  9%] Compiling ==> thirdparty/assimp/code/JoinVerticesProcess.cpp
[  9%] Compiling ==> thirdparty/assimp/code/LimitBoneWeightsProcess.cpp
[  9%] Compiling ==> thirdparty/assimp/code/MMDImporter.cpp
[  9%] Compiling ==> thirdparty/assimp/code/MMDPmxParser.cpp
[  9%] Compiling ==> thirdparty/assimp/code/MakeVerboseFormat.cpp
[  9%] Compiling ==> thirdparty/assimp/code/MaterialSystem.cpp
[  9%] Compiling ==> thirdparty/assimp/code/OptimizeGraph.cpp
[  9%] Compiling ==> thirdparty/assimp/code/OptimizeMeshes.cpp
[  9%] Compiling ==> thirdparty/assimp/code/PostStepRegistry.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/PretransformVertices.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/ProcessHelper.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/RawLoader.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/RemoveComments.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/RemoveRedundantMaterials.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/RemoveVCProcess.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SGSpatialSort.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/ScaleProcess.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SceneCombiner.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/ScenePreprocessor.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SkeletonMeshBuilder.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SortByPTypeProcess.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SpatialSort.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SplitByBoneCountProcess.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/SplitLargeMeshes.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/StandardShapes.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/Subdivision.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/TargetAnimation.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/TextureTransform.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/TriangulateProcess.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/ValidateDataStructure.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/Version.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/VertexTriangleAdjacency.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/scene.cpp
[ 10%] Compiling ==> thirdparty/assimp/code/simd.cpp
[ 10%] Compiling ==> modules/assimp/editor_scene_importer_assimp.cpp
[ 11%] Compiling ==> modules/assimp/register_types.cpp
[ 11%] Compiling ==> modules/bmp/image_loader_bmp.cpp
[ 11%] Compiling ==> modules/bmp/register_types.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btAxisSweep3.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btBroadphaseProxy.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btCollisionAlgorithm.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btDbvt.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btDbvtBroadphase.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btDispatcher.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btOverlappingPairCache.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btQuantizedBvh.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/BroadphaseCollision/btSimpleBroadphase.cpp
[ 11%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btActivatingCollisionAlgorithm.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btBoxBoxCollisionAlgorithm.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btBox2dBox2dCollisionAlgorithm.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btBoxBoxDetector.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionDispatcher.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionDispatcherMt.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionObject.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionWorld.cpp
[ 12%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCollisionWorldImporter.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCompoundCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btCompoundCompoundCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btConvexConcaveCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btConvexConvexAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btConvexPlaneCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btConvex2dConvex2dAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btDefaultCollisionConfiguration.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btEmptyCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btGhostObject.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btHashedSimplePairCache.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btInternalEdgeUtility.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btManifoldResult.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btSimulationIslandManager.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btSphereBoxCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btSphereSphereCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btSphereTriangleCollisionAlgorithm.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/btUnionFind.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionDispatch/SphereTriangleDetector.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btBoxShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btBox2dShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btBvhTriangleMeshShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btCapsuleShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btCollisionShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btCompoundShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConcaveShape.cpp
[ 13%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConeShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvexHullShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvexInternalShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvexPointCloudShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvexPolyhedron.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvexShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvex2dShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btConvexTriangleMeshShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btCylinderShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btEmptyShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btHeightfieldTerrainShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btMinkowskiSumShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btMiniSDF.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btMultimaterialTriangleMeshShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btMultiSphereShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btOptimizedBvh.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btPolyhedralConvexShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btScaledBvhTriangleMeshShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btSdfCollisionShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btShapeHull.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btSphereShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btStaticPlaneShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btStridingMeshInterface.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTetrahedronShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTriangleBuffer.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTriangleCallback.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTriangleIndexVertexArray.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTriangleIndexVertexMaterialArray.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTriangleMesh.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btTriangleMeshShape.cpp
[ 14%] Compiling ==> thirdparty/bullet/BulletCollision/CollisionShapes/btUniformScalingShape.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btContactProcessing.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btGenericPoolAllocator.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btGImpactBvh.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btGImpactCollisionAlgorithm.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btGImpactQuantizedBvh.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btGImpactShape.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/btTriangleShapeEx.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/gim_box_set.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/gim_contact.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/gim_memory.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/Gimpact/gim_tri_collision.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btContinuousConvexCollision.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btConvexCast.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btGjkConvexCast.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btGjkEpa2.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btGjkEpaPenetrationDepthSolver.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btGjkPairDetector.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btMinkowskiPenetrationDepthSolver.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btPersistentManifold.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btRaycastCallback.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btSubSimplexConvexCast.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btVoronoiSimplexSolver.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletCollision/NarrowPhaseCollision/btPolyhedralContactClipping.cpp
[ 15%] Compiling ==> thirdparty/bullet/BulletDynamics/Character/btKinematicCharacterController.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btConeTwistConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btContactConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btFixedConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btGearConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btGeneric6DofConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btGeneric6DofSpringConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btGeneric6DofSpring2Constraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btHinge2Constraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btHingeConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btPoint2PointConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btSequentialImpulseConstraintSolver.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btNNCGConstraintSolver.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btSliderConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btSolve2LinearConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btTypedConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/ConstraintSolver/btUniversalConstraint.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/Dynamics/btDiscreteDynamicsWorld.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/Dynamics/btDiscreteDynamicsWorldMt.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/Dynamics/btSimulationIslandManagerMt.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/Dynamics/btRigidBody.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/Dynamics/btSimpleDynamicsWorld.cpp
[ 16%] Compiling ==> thirdparty/bullet/BulletDynamics/Vehicle/btRaycastVehicle.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Vehicle/btWheelInfo.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBody.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyConstraintSolver.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyJointLimitConstraint.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyConstraint.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyPoint2Point.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyFixedConstraint.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodySliderConstraint.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyJointMotor.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/Featherstone/btMultiBodyGearConstraint.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/MLCPSolvers/btDantzigLCP.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/MLCPSolvers/btMLCPSolver.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletDynamics/MLCPSolvers/btLemkeAlgorithm.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletInverseDynamics/IDMath.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletInverseDynamics/MultiBodyTree.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletInverseDynamics/details/MultiBodyTreeInitCache.cpp
[ 17%] Compiling ==> thirdparty/bullet/BulletInverseDynamics/details/MultiBodyTreeImpl.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftBody.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftBodyConcaveCollisionAlgorithm.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftBodyHelpers.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftBodyRigidBodyCollisionConfiguration.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftRigidCollisionAlgorithm.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftRigidDynamicsWorld.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftMultiBodyDynamicsWorld.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btSoftSoftCollisionAlgorithm.cpp
[ 18%] Compiling ==> thirdparty/bullet/BulletSoftBody/btDefaultSoftBodySolver.cpp
[ 18%] Compiling ==> thirdparty/bullet/clew/clew.c
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btAlignedAllocator.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btConvexHull.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btConvexHullComputer.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btGeometryUtil.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btPolarDecomposition.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btQuickprof.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btSerializer.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btSerializer64.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btThreads.cpp
[ 18%] Compiling ==> thirdparty/bullet/LinearMath/btVector3.cpp
[ 18%] Compiling ==> modules/bullet/area_bullet.cpp
In file included from modules/bullet/area_bullet.cpp:31:
In file included from modules/bullet/area_bullet.h:34:
In file included from modules/bullet/collision_object_bullet.h:40:
In file included from thirdparty/bullet/LinearMath/btTransform.h:18:
In file included from thirdparty/bullet/LinearMath/btMatrix3x3.h:18:
thirdparty/bullet/LinearMath/btVector3.h:335:7: error: argument value 10880 is
      outside the valid range [0, 255] [-Wargument-outside-range]
                y = bt_splat_ps(y, 0x80);
                    ^~~~~~~~~~~~~~~~~~~~
thirdparty/bullet/LinearMath/btVector3.h:43:29: note: expanded from macro
      'bt_splat_ps'
#define bt_splat_ps(_a, _i) bt_pshufd_ps((_a), BT_SHUFFLE(_i, _i, _i, _i))
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/bullet/LinearMath/btVector3.h:41:33: note: expanded from macro
      'bt_pshufd_ps'
#define bt_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/xmmintrin.h:2608:11: note: 
      expanded from macro '_mm_shuffle_ps'
  (__m128)__builtin_ia32_shufps((__v4sf)(__m128)(a), (__v4sf)(__m128)(b), \
          ^
In file included from modules/bullet/area_bullet.cpp:31:
In file included from modules/bullet/area_bullet.h:34:
In file included from modules/bullet/collision_object_bullet.h:40:
In file included from thirdparty/bullet/LinearMath/btTransform.h:18:
thirdparty/bullet/LinearMath/btMatrix3x3.h:910:14: error: argument value 10880
      is outside the valid range [0, 255] [-Wargument-outside-range]
        __m128 vk = bt_splat_ps(_mm_load_ss((float*)&k), 0x80);
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/bullet/LinearMath/btVector3.h:43:29: note: expanded from macro
      'bt_splat_ps'
#define bt_splat_ps(_a, _i) bt_pshufd_ps((_a), BT_SHUFFLE(_i, _i, _i, _i))
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
thirdparty/bullet/LinearMath/btVector3.h:41:33: note: expanded from macro
      'bt_pshufd_ps'
#define bt_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.0/include/xmmintrin.h:2608:11: note: 
      expanded from macro '_mm_shuffle_ps'
  (__m128)__builtin_ia32_shufps((__v4sf)(__m128)(a), (__v4sf)(__m128)(b), \
          ^
2 errors generated.
scons: *** [modules/bullet/area_bullet.osx.tools.64.o] Error 1
scons: building terminated because of errors.
```

--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -27,6 +27,9 @@ Files extracted from upstream source:
 - src/* apart from CMakeLists.txt and premake4.lua files
 - LICENSE.txt
 
+Upstream patches included in the `patches` directory have been applied
+on top of the 2.88 source (not included in any stable release yet).
+
 
 ## certs
 

--- a/thirdparty/bullet/Bullet3Common/b3Vector3.h
+++ b/thirdparty/bullet/Bullet3Common/b3Vector3.h
@@ -36,7 +36,7 @@ subject to the following restrictions:
 #pragma warning(disable : 4556)  // value of intrinsic immediate argument '4294967239' is out of range '0 - 255'
 #endif
 
-#define B3_SHUFFLE(x, y, z, w) ((w) << 6 | (z) << 4 | (y) << 2 | (x))
+#define B3_SHUFFLE(x, y, z, w) (((w) << 6 | (z) << 4 | (y) << 2 | (x)) & 0xff)
 //#define b3_pshufd_ps( _a, _mask ) (__m128) _mm_shuffle_epi32((__m128i)(_a), (_mask) )
 #define b3_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
 #define b3_splat3_ps(_a, _i) b3_pshufd_ps((_a), B3_SHUFFLE(_i, _i, _i, 3))

--- a/thirdparty/bullet/LinearMath/btVector3.h
+++ b/thirdparty/bullet/LinearMath/btVector3.h
@@ -36,7 +36,7 @@ subject to the following restrictions:
 #pragma warning(disable : 4556)  // value of intrinsic immediate argument '4294967239' is out of range '0 - 255'
 #endif
 
-#define BT_SHUFFLE(x, y, z, w) ((w) << 6 | (z) << 4 | (y) << 2 | (x))
+#define BT_SHUFFLE(x, y, z, w) (((w) << 6 | (z) << 4 | (y) << 2 | (x)) & 0xff)
 //#define bt_pshufd_ps( _a, _mask ) (__m128) _mm_shuffle_epi32((__m128i)(_a), (_mask) )
 #define bt_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
 #define bt_splat3_ps(_a, _i) bt_pshufd_ps((_a), BT_SHUFFLE(_i, _i, _i, 3))

--- a/thirdparty/bullet/patches/bullet-git-fix-build-clang-8.patch
+++ b/thirdparty/bullet/patches/bullet-git-fix-build-clang-8.patch
@@ -1,0 +1,36 @@
+From 7638b7c5a659dceb4e580ae87d4d60b00847ef94 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Emil=20Nord=C3=A9n?= <emilnorden@yahoo.se>
+Date: Sat, 4 May 2019 08:38:53 +0200
+Subject: [PATCH] fixed build on latest version of clang
+
+---
+ src/Bullet3Common/b3Vector3.h | 2 +-
+ src/LinearMath/btVector3.h    | 2 +-
+ 2 files changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/src/Bullet3Common/b3Vector3.h b/src/Bullet3Common/b3Vector3.h
+index 56e6c13311..a70d68d6e1 100644
+--- a/src/Bullet3Common/b3Vector3.h
++++ b/src/Bullet3Common/b3Vector3.h
+@@ -36,7 +36,7 @@ subject to the following restrictions:
+ #pragma warning(disable : 4556)  // value of intrinsic immediate argument '4294967239' is out of range '0 - 255'
+ #endif
+ 
+-#define B3_SHUFFLE(x, y, z, w) ((w) << 6 | (z) << 4 | (y) << 2 | (x))
++#define B3_SHUFFLE(x, y, z, w) (((w) << 6 | (z) << 4 | (y) << 2 | (x)) & 0xff)
+ //#define b3_pshufd_ps( _a, _mask ) (__m128) _mm_shuffle_epi32((__m128i)(_a), (_mask) )
+ #define b3_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
+ #define b3_splat3_ps(_a, _i) b3_pshufd_ps((_a), B3_SHUFFLE(_i, _i, _i, 3))
+diff --git a/src/LinearMath/btVector3.h b/src/LinearMath/btVector3.h
+index 61fd8d1e46..d65ed9808d 100644
+--- a/src/LinearMath/btVector3.h
++++ b/src/LinearMath/btVector3.h
+@@ -36,7 +36,7 @@ subject to the following restrictions:
+ #pragma warning(disable : 4556)  // value of intrinsic immediate argument '4294967239' is out of range '0 - 255'
+ #endif
+ 
+-#define BT_SHUFFLE(x, y, z, w) ((w) << 6 | (z) << 4 | (y) << 2 | (x))
++#define BT_SHUFFLE(x, y, z, w) (((w) << 6 | (z) << 4 | (y) << 2 | (x)) & 0xff)
+ //#define bt_pshufd_ps( _a, _mask ) (__m128) _mm_shuffle_epi32((__m128i)(_a), (_mask) )
+ #define bt_pshufd_ps(_a, _mask) _mm_shuffle_ps((_a), (_a), (_mask))
+ #define bt_splat3_ps(_a, _i) bt_pshufd_ps((_a), BT_SHUFFLE(_i, _i, _i, 3))

